### ISSUE TITLE
Disable password logins when oauth2 is used

### ISSUE THREAD

**Issue by luna-duclos** [go-gitea/gitea#25393]
### Feature Description

It would be great if gitea would allow me to disable password changes for users who's account has been created via an oauth2 login and disallow them to login using it even if they had set it.

As well as hide the relevant components of the web UI.

a PR for this was made in the past: #19344, but it seems it was closed without review. I was wondering if it would be something that is ok to potentially recontribute ?

**Comment by lunny** [comment]
Should that be an option of this OAuth2 source setting?

**Comment by silverwind** [comment]
Has to be a global config option I assume as there can be multiple oauth sources etc.

**Comment by wxiaoguang** [comment]
I do not think this issue can be resolved by "easy" patches.

See 

* https://github.com/go-gitea/gitea/issues/24821

> The first step IMO is to have a (nearly) complete design,to define the expected behaviors for various situations.

**Comment by luna-duclos** [comment]
> Has to be a global config option I assume as there can be multiple oauth sources etc. Also, it is quite a risky thing because if oauth provider is down, not even admin can log in.

As long as its an app.ini setting, whoever administers the instance could disable it again to log in as admin again, no ?

**Comment by jaltgen** [comment]
I'd be in favor of this as well, Portainer has this option in the GUI, for example 👍 I understand it may not fit in the current scope of work or schedule, its just one of those small things that may increase security and adoption with organisations?

**Comment by gtherond** [comment]
I'd like to second this as we exactly use that with our on-premise coder (coder.com) instance and it works like a bliss as we can still enable back the password based login by just switching back the environment variable to its default value.

This allows us (as long as with the disabled signup feature) to tightly control who access and sign in the forge.

Additionally, we're using a kubernetes deployment so switching back some env vars is swift and smooth without any visible interrupt for our users.

**Comment by phluxx** [comment]
Third for this request

**Comment by Edo78** [comment]
I just installed gitea and I'm working through the configuration options to have it tailored to my needs.
I found
```
ENABLE_BASIC_AUTHENTICATION: true: Disable this to disallow authentication using HTTP BASIC and the user's password. Please note if you disable this you will not be able to access the tokens API endpoints using a password. Further, this only disables BASIC authentication using the password - not tokens or OAuth Basic.

ENABLE_PASSWORD_SIGNIN_FORM: true: Show the password login form (for password-based login). If you set it to false, maybe it also needs to set ENABLE_BASIC_AUTHENTICATION to false to completely disable password-based authentication.
```
but even with both set to `false` the login form is still there.
Then I came here searching for an answer but, while writing this comment, I noticed the latest helm chart is for version 1.22.3 while the latest gitea release is the 1.23.1 ... I forced the chart to use the latest release and the two options worked like a charm and the login form vanished ...

... so why is this issue still open? Am I missing something?

**Comment by wxiaoguang** [comment]
Yes, I think this issue has been resolved (there are too many related issues, this one is missing to close)

Thank you for your feedback!
